src/uri/include/sourcemeta/jsontoolkit/uri.h

```diff
@@ -308,6 +310,9 @@ class SOURCEMETA_JSONTOOLKIT_URI_EXPORT URI {
   [[nodiscard]] auto userinfo() const -> std::optional<std::string_view>;
 
 private:
+  bool parsed = false;
```

Comment:
Can we add a TODO item to remove this `parsed` member eventually? I think there are ways to do it without it and that will look a bit cleaner

Reply:
Yeah I don't like it, but until we have migrate all other method to use our internal structure I prefer to keep it. Lets add the todo.